src/Symfony/Component/Validator/Constraint.php

```diff
@@ -231,10 +232,16 @@ public function __get($option)
      * @param string $group
      *
      * @api
+     *
+     * @throws GroupDefinitionException If the implicit group has been explicitly configured in current constraint.
      */
     public function addImplicitGroupName($group)
     {
-        if (in_array(self::DEFAULT_GROUP, $this->groups) && !in_array($group, $this->groups)) {
+        if (in_array($group, $this->groups)) {
+            throw new GroupDefinitionException(sprintf('The implicit group "%s" has already been explicitly configured in constraint %s', $group, get_class($this)));
```

Comment:
Although this error is technically correct, it isn't very helpful. I would reword the message so that developers hitting the error have the most benefit. They will hit the error when they define the implicit group name themselves, so they should remove it or use some other group instead. This error message should reflect that IMO.